src/views/ToDoList/ToDoList.test.tsx

```diff
@@ -0,0 +1,262 @@
+      beforeEach(() => {
+        (toDosApi.fetchToDos as Mock).mockResolvedValue(returnedTasksList);
+        (toDosApi.createNewTask as Mock).mockReturnValue(
+          new Promise(() => null),
+        );
+      });
+      it('should make attempt to create the POST request', () => {
+        const toDoList = render(<ToDoList />);
+
+        const titleInput = toDoList.getByPlaceholderText('title');
+
+        fireEvent.input(titleInput, { target: { value: newTaskTitle } });
+
+        const button = toDoList.getByRole('button');
+
+        fireEvent.click(button);
+
+        expect(toDosApi.createNewTask).toHaveBeenCalledWith(newTaskMock);
+      });
+      describe('if the Api call is successful', () => {
+        beforeEach(() => {
+          (toDosApi.createNewTask as Mock).mockResolvedValue(newTaskMock);
+        });
+        it('should create new task and not throw error', async () => {
+          const toDoList = render(<ToDoList />);
+
+          const titleInput = toDoList.getByPlaceholderText('title');
+
+          fireEvent.change(titleInput, { target: { value: newTaskTitle } });
+
+          const button = toDoList.getByRole('button');
+
+          fireEvent.click(button);
+
+          const newTaskParagraph = await toDoList.findByText(newTaskTitle, {
+            selector: 'p',
+          });
+
+          const errorParagraph = toDoList.queryByText(
+            'Failed to catch tasks list',
+            { selector: 'p' },
+          );
+
+          expect(newTaskParagraph).toBeDefined();
+          expect(errorParagraph).toBeNull();
+        });
+      });
+      describe('if the Api call is unsuccessful', () => {
+        beforeEach(() => {
+          (toDosApi.createNewTask as Mock).mockRejectedValue(null);
+        });
+        it('should throw an error and not create task', async () => {
+          const toDoList = render(<ToDoList />);
+
+          const titleInput = toDoList.getByPlaceholderText('title');
+
+          fireEvent.change(titleInput, { target: { value: newTaskTitle } });
+
+          const button = toDoList.getByRole('button');
+
+          fireEvent.click(button);
+
+          const newTaskParagraph = toDoList.queryByText(newTaskTitle, {
+            selector: 'p',
+          });
+
+          const errorParagraph = await toDoList.findByText(
+            'Something went wrong when creating the task',
+            { selector: 'p' },
+          );
+
+          expect(newTaskParagraph).toBeNull();
+          expect(errorParagraph).toBeDefined();
+        });
+      });
+    });
+  });
+  describe('when user clicks delete button on specific task', () => {
+    beforeEach(() => {
+      (toDosApi.fetchToDos as Mock).mockResolvedValue(returnedTasksList);
+    });
+    describe('and the Api call is successful', () => {
+      beforeEach(() => {
+        (toDosApi.deleteTask as Mock).mockResolvedValue('200');
+      });
+      it('should delete specific task and leave other tasks intact', async () => {
```

Comment:
- deleting the specific task
- leaving other tasks intact
are two separate test cases and should have two separate `it`s